src/Discord/Internal/Types/Interactions.hs

```diff
@@ -0,0 +1,363 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Discord.Internal.Types.Interactions
+  ( Interaction (..),
+    InteractionId,
+    InteractionToken,
+    InteractionType (..),
+    InteractionData (..),
+    ResolvedData (..),
+    ApplicationCommandInteractionDataOption (..),
+    InteractionResponse (..),
+    InteractionCallbackType (..),
+    InteractionCallbackData (..),
+    InteractionCallbackAutocomplete,
+    InteractionCallbackMessages (..),
+    InteractionCallbackDataFlags (..),
+    InteractionCallbackDataFlag (..),
+  )
+where
+
+import Data.Aeson
+import Data.Bits (Bits (shift, (.|.)))
+import Data.Data ( Data(dataTypeOf), dataTypeConstrs, fromConstr )
+import Data.Default (Default (..))
+import Data.Maybe (fromJust)
+import Data.Text (Text)
+import Discord.Internal.Types.ApplicationCommands
+import Discord.Internal.Types.Channel (AllowedMentions, Attachment, Message)
+import Discord.Internal.Types.Embed (Embed)
+import Discord.Internal.Types.Guild (GuildMember)
+import Discord.Internal.Types.Prelude (ApplicationId, ChannelId, GuildId, Snowflake)
+import Discord.Internal.Types.User (User)
+
+toMaybeJSON :: (ToJSON a) => a -> Maybe Value
+toMaybeJSON = return . toJSON
+
+makeTable :: (Data t, Enum t) => t -> [(Int, t)]
+makeTable t = map (\cData -> let c = fromConstr cData in (fromEnum c, c)) (dataTypeConstrs $ dataTypeOf t)
+
+type InteractionId = Snowflake
```

Comment:
Move to internal.types.prelude

Reply:
Just the type, or the `makeTable` too?

Reply:
Oh true, just the type, since `makeTable` is a utility only used in this file.